components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -129,61 +251,85 @@ public void destroy() {
         String to;
         String subject;
         String body;
+        String type;
 
-        EmailSender(String to, String subject, String body) {
+        EmailSender(String to, String subject, String body, String type) {
             this.to = to;
             this.subject = subject;
             this.body = body;
+            this.type = type;
         }
 
+        /**
+         * Sending emails to the corresponding Email IDs'.
+         */
+
         @Override
         public void run() {
-            Map<String, String> headerMap = new HashMap<String, String>();
-            headerMap.put(MailConstants.MAIL_HEADER_SUBJECT, subject);
-            OMElement payload = OMAbstractFactory.getOMFactory().createOMElement(
-                    BaseConstants.DEFAULT_TEXT_WRAPPER, null);
-            payload.setText(body);
 
-            ServiceClient serviceClient = null;
+            if (log.isDebugEnabled()) {
+                log.debug("Format of the email:" + " " + to + "->" + type);
+            }
+
+
+            //Creating MIME object using initiated session.
+
+            MimeMessage message = new MimeMessage(session);
+
+            StringBuilder stringBuilder = new StringBuilder();
+
+            if (type.equals(EmailEventAdapterConstants.MAIL_TEXT_PLAIN)) {
+
+                //parsing only text as the email body.
+
+                stringBuilder.append(body);
+
+            } else if (type.equals(EmailEventAdapterConstants.MAIL_TEXT_HTML)) {
+
+                //Append desired HTML tags to the email body here.
+
+                stringBuilder.append("<b>");
+                stringBuilder.append("<u>");
+                stringBuilder.append(body);
+                stringBuilder.append("</u>");
+                stringBuilder.append("</b>");
+            }
```

Comment:
I just added the HTML tags thinking that users can explicitly modify and define their HTML preference there. I'll remove the if else condition, since the users are sending the HTML code explicitly by their self (no need of adding additional HTML and Hard coded HTML elements).